pint/models/dispersion_model.py

```diff
@@ -0,0 +1,128 @@
+"""This module implements a simple model of a constant dispersion measure.
+   And DMX dispersion"""
+# dispersion.py
+# Simple (constant) ISM dispersion measure
+from warnings import warn
+import parameter as p
+from .timing_model import TimingModel, Cache
+import astropy.units as u
+import numpy as np
+import pint.utils as ut
+import astropy.time as time
+# The units on this are not completely correct
+# as we don't really use the "pc cm^3" units on DM.
+# But the time and freq portions are correct
+DMconst = 1.0/2.41e-4 * u.MHz * u.MHz * u.s
+
+class Dispersion(TimingModel):
+    """This class provides a base dispersion timing model. The dm varience will
+    be treated linearly.
+    """
+    def __init__(self):
+        super(Dispersion, self).__init__()
+        self.add_param(p.floatParameter(name="DM",
+                       units="pc cm^-3", value=0.0,
+                       description="Dispersion measure"))
+        self.dm_value_funcs = [self.constant_dm,]
+        self.delay_funcs['L1'] += [self.dedispersion_delay,]
+
+    def setup(self):
+        super(Dispersion, self).setup()
+
+    def constant_dm(self, toas):
+        cdm = np.zeros(len(toas))
+        cdm.fill(self.DM.num_value)
+        return cdm * self.DM.num_unit
+
+    def dispersion_time_delay(self, DM, freq):
+        """Return the dispersion time delay for a set of frequency."""
+        # dm delay
+        dmdelay = DM * DMconst / freq**2.0
```

Comment:
Both this line and the DMconst definition above should have a reference to the literature for where the equation and value come from, such as Eqns 4.4 and 4.6 from Lorimer and Kramer.  In fact, looking at that, I think the DMconst should be DMconst = 4148.808.  Also, we should include a note that the equation assumes that freq >> f_p (the plasma freq)